docs/developer.md

```diff
@@ -0,0 +1,25 @@
+# Developer Guide
```

Comment:
We should probably have 2 sections:
- devcontainer
- "classic" (currently covered in your doc)